### PR TITLE
MT: Implemented 68-97, fixes on 111

### PR DESCRIPTION
**Pull request by starg09**
TODO:
* Write the check for Vigoroth (68) to increase the damage in "Wake-Up Punch".
Fixed:
* Searches should now work correctly on Fossil Excavator (111).

### CODE REVIEW

File: src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy [hunk 1]
```diff
@@ -1607,15 +1611,20 @@ public enum MysteriousTreasures implements LogicCardInfo {
             energyCost P
             attackRequirement {}
             onAttack {
-              damage 0
+              my.deck.subList(0,5).select(min:0,"Choose a card to put in your hand").moveTo(hidden: true, my.hand)
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy:1614]: Are you supposed to be able to choose 0? The text implies to me that you _must_ select 1.

File: src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy [hunk 2]
```diff
@@ -1607,15 +1611,20 @@ public enum MysteriousTreasures implements LogicCardInfo {
             energyCost P
             attackRequirement {}
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy:1612]: Probably needs a deck assertion too.

File: src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy (hunk 1 above)
Comment (starg09) [reply src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy:1614]: Yup, should be 1 indeed. Will correct it with the build fix 👍

File: src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy [hunk 3]
```diff
@@ -1660,32 +1672,39 @@ public enum MysteriousTreasures implements LogicCardInfo {
...
-            text "Any damage. "
-            energyCost ()
-            attackRequirement {}
-            onAttack {
-              damage 0
+          //Wacan Berry: Any damage done to Bidoof by attacks from [L] Pokémon is reduced by 20 (after applying Weakness and Resistance).
+          customAbility {
+            delayedA {
+              before APPLY_ATTACK_DAMAGES, {
+                if(ef.attacker.owner != self.owner && (ef.attacker.types.contains(R) || ef.attacker.types.contains(W))) {
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy:1691]: Shouldn't this just be checking for `L` type?

File: src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy [hunk 4]
```diff
@@ -1660,32 +1672,39 @@ public enum MysteriousTreasures implements LogicCardInfo {
...
+                  bg.dm().each{
+                    if(it.to == self && it.notNoEffect && it.dmg.value) {
+                      bc "Chesto Berry -20"
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy:1694]: I'm guessing this whole thing was copy and pasted from another card and you forgot to change it :laughing:

File: src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy (hunk 3 above)
Comment (starg09) [reply src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy:1691]: Ah, forgot to edit the check whoops. Thanks!

File: src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy (hunk 4 above)
Comment (starg09) [reply src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy:1694]: Replied above 😝

File: src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy [hunk 5]
```diff
@@ -1993,7 +2044,8 @@ public enum MysteriousTreasures implements LogicCardInfo {
             energyCost W, W
             attackRequirement {}
             onAttack {
-              damage 0
+              //TODO: Modularize
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy:2047]: Something like

```
def doDamage = true
flip 2, {}, {doDamage = false}
if(doDamage) damage 120
```
should work as well, for when you modularize this, right?

File: src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy [hunk 6]
```diff
@@ -2024,11 +2074,11 @@ public enum MysteriousTreasures implements LogicCardInfo {
         return basic (this, hp:HP050, type:PSYCHIC, retreatCost:1) {
           weakness P, PLUS10
           move "Call for Family", {
-            text "Search your deck for Nidoran Male or Nidoran Female and put it onto your Bench. Shuffle your deck afterward."
+            text "Search your deck for Nidoran ♂ or Nidoran ♀ and put it onto your Bench. Shuffle your deck afterward."
             energyCost C
 
-            // TODO
-            callForFamily(basic:true, 1, delegate)
+            //TODO: Check this finding all Nidorans, male and female.
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy:2080]: This should work fine based of how `callForFamily` works

**Pull request merged**